### PR TITLE
Connect to an Azure Service Bus namespace using an Azure Private Endpoint

### PR DESCRIPTION
**Pull request by paolosalvatori**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* First commit

### CODE REVIEW

File: 201-servicebus-private-endpoint/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+## Architecture ##
+
+The following picture shows the architecture and network topology of the sample.
+
+![Architecture](images/architecture.png)
+
+The ARM template deploys the following resources:
+
+- Virtual Network: this virtual network has a single subnet that hosts an Linux (Ubuntu) virtual machine
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/README.md:26]: "an Linux" should be "a Linux"

File: 201-servicebus-private-endpoint/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+- Network Security Group: this resource contains an inbound rule to allow the access to the virtual machine on port 22 (SSH)
+- The virtual machine is created with a managed identity which is assigned the contributor role at the resource group scope level
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/README.md:28]: why is this needed?

File: 201-servicebus-private-endpoint/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+The ARM template uses the [Azure Custom Script Extension](https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-linux) to download and run the following Bash script on the virtual machine. The script performs the following steps:
+
+- Validates the parameters received by the Custom Script extension
+- Updates the system and upgrade packages
+- Runs the nslookup command against the public URL of the Storage Account to verify that this gets resolved to a private address
+- Runs the nslookup command against the public URL of the Service Bus namespace to verify that this gets resolved to a private address
+
+**servicebus_nslookup.sh**
+
+```bash
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/README.md:49]: can we just link to the script instead of duplicating inline?  Will create a maint issue otherwise

File: 201-servicebus-private-endpoint/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+nslookup $serviceBusNamespaceEndpoint
+
+# Run nslookup to verify that public hostname of the Blob storage account 
+# is properly mapped to the private address of the provate endpoint
+nslookup $blobServicePrimaryEndpoint
+```
+
+## Deployment ##
+
+You can use the template.json ARM template and parameters.json file included in this repository to deploy the sample. Make sure to edit the parameters.json file to customize the installation. You can also use the deploy.sh Bash script under the scripts folder to deploy the ARM template. The following figure shows the resources deployed by the ARM template in the target resource group.
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/README.md:87]: just remove this section since it's the same for every sample in the repo (and it's not correct ;))

File: 201-servicebus-private-endpoint/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,1040 @@
...
+      "value": "[variables('scriptFileUri')]",
+      "type": "string"
+    },
+    "environment": {
+      "value": "[environment()]",
+      "type": "object"
+    },
+    "NamespaceConnectionString": {
+      "type": "string",
+      "value": "[listkeys(variables('authRuleResourceId'), '2017-04-01').primaryConnectionString]"
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/azuredeploy.json]: remove secrets from outputs

File: 201-servicebus-private-endpoint/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,1040 @@
...
+        "sshPublicKey",
+        "password"
+      ],
+      "metadata": {
+        "description": "Specifies the type of authentication when accessing the Virtual Machine. SSH key is recommended."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "defaultValue": "azadmin",
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/azuredeploy.json:116]: remove defaultValue (increases surface area)

File: 201-servicebus-private-endpoint/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,1040 @@
...
+    "serviceBusNamespacePrivateEndpoint": {
+      "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', parameters('serviceBusNamespacePrivateEndpointName')), '2020-04-01', 'Full')]",
+      "type": "object"
+    },
+    "blobStorageAccountPrivateEndpoint": {
+      "value": "[reference(resourceId('Microsoft.Network/privateEndpoints', parameters('blobStorageAccountPrivateEndpointName')), '2020-04-01', 'Full')]",
+      "type": "object"
+    },
+    "serviceBusNamespace": {
+      "value": "[reference(resourceId('Microsoft.ServiceBus/namespaces', parameters('serviceBusNamespaceName')), '2017-04-01', 'Full')]",
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/azuredeploy.json]: update apiVersion

File: 201-servicebus-private-endpoint/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,1040 @@
...
+    "serviceBusNamespaceEndpoint": "[concat(parameters('serviceBusNamespaceName'), variables('serviceBusPublicDNSZoneForwarder'))]",
+    "blobServicePrimaryEndpoint": "[concat(parameters('blobStorageAccountName'), variables('blobPublicDNSZoneForwarder'))]",
+    "serviceBusNamespacePrivateEndpointId": "[resourceId('Microsoft.Network/privateEndpoints', parameters('serviceBusNamespacePrivateEndpointName'))]",
+    "blobStorageAccountPrivateEndpointId": "[resourceId('Microsoft.Network/privateEndpoints', parameters('blobStorageAccountPrivateEndpointName'))]",
+    "serviceBusNamespacePrivateEndpointSubResource": "namespace",
+    "blobStorageAccountPrivateEndpointSubResource": "blob"
+  },
+  "resources": [
+    {
+      "apiVersion": "2017-04-01",
```
Comment (bmoore-msft) [inline 201-servicebus-private-endpoint/azuredeploy.json]: updata apiVersion

File: 201-servicebus-private-endpoint/README.md (hunk 2 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/README.md:28]: Copy and paste from the README of a project on my GitHub account. I removed it.

File: 201-servicebus-private-endpoint/README.md (hunk 1 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/README.md:26]: Fixed thanks

File: 201-servicebus-private-endpoint/README.md (hunk 3 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/README.md:49]: Sounds good. Fixed.

File: 201-servicebus-private-endpoint/README.md (hunk 4 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/README.md:87]: I did not remove the section as the image is different :) The wording is the same, the picture of the resources in the resource group are different :)

File: 201-servicebus-private-endpoint/azuredeploy.json (hunk 6 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/azuredeploy.json:116]: Done

File: 201-servicebus-private-endpoint/azuredeploy.json (hunk 5 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/azuredeploy.json]: Removed

File: 201-servicebus-private-endpoint/azuredeploy.json (hunk 7 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/azuredeploy.json]: I'm using the latest apiVersion for Service Bus: 2018-01-01-preview

File: 201-servicebus-private-endpoint/azuredeploy.json (hunk 8 above)
Comment (paolosalvatori) [reply 201-servicebus-private-endpoint/azuredeploy.json]: Solved

**Comment by paolosalvatori** [comment]
I made some changes to the previous blob-storage-private-endpoint, multiple commits, and now the ADO pipelines fail as they try to run from the wrong folder. I'll create another PR.
